### PR TITLE
Adoption to firewall-controller-manager API `firewall.metal-stack.io/v2`

### LINKED ISSUE #10: Network Traffic should not count specific connections
It must be possible to exclude network traffic which is partition local for example.
This can be achieved by switching network accounting from a interface based accounting to nftables based counters. We must then create counter rules for "ignored" networks and all other networks per vrf id.

The list if ignored networks is stored in the firewall crd. This is done from the gardener-extension-provider-metal which will ask metal-api to get these networks.

@mwennrich @mwindower

### LINKED ISSUE #104: Controller download URL must be secured by signature
Probably needs a second signature for migration.

Maybe we could also think about introducing a validation webhook, which can be deployed into the shoot namespace of the seed cluster right next to customer's API server.

A validation webhook could also make very useful checks for other purposes, too:

- Ensure `to` section is present in egress policy

### PR DESCRIPTION
**Pull request by majst01**
- Is fully dependent on the new firewall-controller-manager being in place, must be a next major release
- Changes to new architecture: Seed and Shoot client
- Integration test was removed, because it was quite complex and it does not do a lot --> we need to rethink the integration tests and create new, more complete ones in another pull request

Closes #104. Closes #10.

References #35.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -76,6 +80,7 @@ manifests: controller-gen fetch-template
 
 # Fetch firewall template
 fetch-template:
+	# FIXME: If this is embedded into the networker, why not just use from embedded source?
 	wget https://raw.githubusercontent.com/metal-stack/metal-networker/${METAL_NETWORKER_VERSION}/pkg/netconf/tpl/frr.firewall.tpl -O ./pkg/network/frr.firewall.tpl
```
Comment (majst01) [inline Makefile:84]: agree

**Comment by Gerrit91** [comment]
Upload currently failing for some reason, seems to be a tracked issue already: https://github.com/google-github-actions/setup-gcloud/issues/571

File: Dockerfile [hunk 2]
```diff
@@ -1,42 +1,6 @@
...
-
-# Install kubenuilder
-ENV KUBEBUILDER_ASSETS=/usr/local/bin
-RUN make kubebuilder
-
-# Build
-RUN make test all
-
-# Final Image
 FROM debian:10
```
Comment (majst01) [inline Dockerfile:1]:
```suggestion
FROM scratch
```

File: Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM debian:10
+FROM scratch
 WORKDIR /
 COPY bin/firewall-controller .
 RUN apt update \
```
Comment (majst01) [inline Dockerfile]: This will not work :-)

File: pkg/nftables/firewall.go [hunk 4]
```diff
@@ -235,7 +236,7 @@ func (f *Firewall) validate(file string) error {
 }
 
 func (f *Firewall) reconcileIfaceAddresses() error {
-	var errors *multierror.Error
+	var errs *multierror.Error
```
Comment (majst01) [inline pkg/nftables/firewall.go:239]: i would prefer to completely get rid of multierr and use errors.Join at the end from the stdlib

**Review by vknabel — APPROVED** [review]
Code review fine from my perspective. Unsure how to test, though

**Comment by majst01** [comment]
> Code review fine from my perspective. Unsure how to test, though. Eventually wait for further feedback from @majst01

I cannot approve because i was the initial creator of the PR :-)

**Pull request merged**